### PR TITLE
FEAT: add google sign in button

### PR DESCRIPTION
**Pull request by vercia**
# Description

- add google sign in logic
- update swagger

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Screenshot(s)

Please attach some screenshot or a movie if provided changes affects UI.

**Test Configuration**:

- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:

- [x] Add correct label to your pull request
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/components/molecules/SocialButtons/GoogleButton/index.tsx [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { isExpoGo, isWeb } from '@baca/constants'
+
+import { NativeGoogleButton } from './NativeGoogleButton'
+
+export const GoogleButton = () => {
+  if (isExpoGo || isWeb) return undefined //TODO: Add google button for web
```
Comment (MateuszRostkowski) [inline src/components/molecules/SocialButtons/GoogleButton/index.tsx:6]: Change `undefined` to `null`

```suggestion
  // TODO: Add google button for web
  if (isExpoGo || isWeb) return null 
```

File: src/components/molecules/SocialButtons/SocialButton.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+type SocialButtonProps = {
+  onPress: () => void
+  type: SocialMediaType
+} & ButtonProps
+
+export const SocialButton: FC<SocialButtonProps> = ({ type = 'google', ...rest }) => {
+  const { colorScheme } = useColorScheme()
+
+  const { source, text } = SocialButtonVariants[type]
```
Comment (MateuszRostkowski) [inline src/components/molecules/SocialButtons/SocialButton.tsx:38]: Shouldn't `socialButtonVariants` be from lower case?

```suggestion
  const { source, text } = socialButtonVariants[type]
```

**Review by MateuszRostkowski — COMMENTED** [review]
@micbaumr Please see my suggestions :D

File: src/components/molecules/SocialButtons/SocialButton.tsx (hunk 2 above)
Comment (micbaumr) [reply src/components/molecules/SocialButtons/SocialButton.tsx:38]: Indeed, its not a type either a component. Good catch ! 👍

**Pull request merged**